layer.py

```diff
@@ -171,16 +192,16 @@ def cloud_action(self):
     def cloud_action(self, action):
         self._cloud_action = action
 
-    def photo_naming(self, field_name: str) -> str:
-        return self._photo_naming.get(
+    def attachment_naming(self, field_name, attachment_type: int) -> str:
```

Comment:
```suggestion
    def attachment_naming(self, field_name, attachment_type: AttachmentType) -> str:
```

I would do the conversion already in the caller.

Reply:
And since this will be annoying to do in the parent, I would simply add a utility function somewhere `get_attachment_type_by_int_value(value: int) -> AttachmentType`.